src/main/java/WayofTime/bloodmagic/api/registry/AlchemyArrayRecipeRegistry.java

```diff
@@ -57,79 +59,110 @@ public static void registerRecipe(ItemStack inputStack, @Nullable ItemStack cata
+
+            if (circleRenderer == null)
+            {
+                recipes.put(ItemStackWrapper.getHolder(inputStack), new AlchemyArrayRecipe(inputStack, catalystStack, arrayEffect, defaultRenderer));
+            } else
+            {
+                recipes.put(ItemStackWrapper.getHolder(inputStack), new AlchemyArrayRecipe(inputStack, catalystStack, arrayEffect, circleRenderer));
+            }
+
             recipes.put(ItemStackWrapper.getHolder(inputStack), new AlchemyArrayRecipe(inputStack, catalystStack, arrayEffect, circleRenderer));
         }
+        else if (input instanceof List)
+        {
+            if (circleRenderer == null)
+            {
+                recipes.put(input, new AlchemyArrayRecipe(input, catalystStack, arrayEffect, defaultRenderer));
+            }
+            else
+            {
+                recipes.put(input, new AlchemyArrayRecipe(input, catalystStack, arrayEffect, circleRenderer));
+            }
+
+            recipes.put(input, new AlchemyArrayRecipe(input, catalystStack, arrayEffect, circleRenderer));
+        }
+    }
 
-        recipes.put(ItemStackWrapper.getHolder(inputStack), new AlchemyArrayRecipe(inputStack, catalystStack, arrayEffect, circleRenderer));
+    public static void registerCraftingRecipe(Object input, ItemStack catalystStack, ItemStack outputStack, AlchemyCircleRenderer circleRenderer)
+    {
+        registerRecipe(input, catalystStack, new AlchemyArrayEffectCrafting(outputStack), circleRenderer);
     }
 
-    public static void registerCraftingRecipe(ItemStack inputStack, ItemStack catalystStack, ItemStack outputStack, AlchemyCircleRenderer circleRenderer)
+    public static void registerCraftingRecipe(Object input, ItemStack catalystStack, ItemStack outputStack, ResourceLocation arrayResource)
     {
-        registerRecipe(inputStack, catalystStack, new AlchemyArrayEffectCrafting(outputStack), circleRenderer);
+        registerRecipe(input, catalystStack, new AlchemyArrayEffectCrafting(outputStack), arrayResource);
     }
 
-    public static void registerCraftingRecipe(ItemStack inputStack, ItemStack catalystStack, ItemStack outputStack, ResourceLocation arrayResource)
+    public static void registerCraftingRecipe(Object input, ItemStack catalystStack, ItemStack outputStack)
     {
-        registerRecipe(inputStack, catalystStack, new AlchemyArrayEffectCrafting(outputStack), arrayResource);
+        registerRecipe(input, catalystStack, new AlchemyArrayEffectCrafting(outputStack));
     }
 
-    public static void registerCraftingRecipe(ItemStack inputStack, ItemStack catalystStack, ItemStack outputStack)
+    public static void registerCraftingRecipe(String inputOreDict, ItemStack catalystStack, ItemStack outputStack)
     {
-        registerRecipe(inputStack, catalystStack, new AlchemyArrayEffectCrafting(outputStack));
+        registerRecipe(OreDictionary.getOres(inputOreDict), catalystStack, new AlchemyArrayEffectCrafting(outputStack));
```

Comment:
Make sure you query 'OreDictionary.doesOreNameExist' before *ever* calling this.